tamr_unify_client/dataset/collection.py

```diff
@@ -171,6 +171,189 @@ def create_from_dataframe(
 
         return dataset
 
+    def create_from_geodataframe(
+        self, geodf, primary_key_name, dataset_name, geo_attr="geometry"
+    ):
+        """Creates a dataset in this collection with the given name, creates an attribute for each column in the `geodf`
+        (with `primary_key_name` as the key attribute), and upserts a record for each row of `geodf`.
+
+        Each attribute has the default type `ARRAY[STRING]`, besides the key attribute, which will have type `STRING`,
+        and the geometry attribute `geo_attr` which will have the type `GEOSPATIAL`.
+
+        This function attempts to ensure atomicity, but it is not guaranteed. If an error occurs while creating
+        attributes or records, an attempt will be made to delete the dataset that was created. However, if this
+        request errors, it will not try again.
+
+        :param geodf: The data to create the dataset with.
```

Comment:
Docstrings should be written in Google-style (see #300 )

Reply:
Updated to google docstring format